packages/suite/src/components/suite/AppNavigation/index.tsx

```diff
@@ -57,21 +57,25 @@ interface Props {
 }
 
 const AppNavigation = ({ items }: Props) => {
-    const routeName = useSelector(state => state.router.route?.name);
+    const router = useSelector(state => state.router);
+    const selectedAccount = useSelector(state => state.wallet.selectedAccount);
     const { goto } = useActions({
         goto: routerActions.goto,
     });
 
+    const { params } = selectedAccount;
     return (
         <Wrapper>
             {items.map(item => {
                 const { route, title, icon, ...restItemProps } = item;
-                const active = routeName === route;
+                const active = router.route?.name === route;
                 return (
                     <StyledNavLink
                         key={route}
                         active={active}
-                        onClick={() => goto(route, undefined, true)}
+                        onClick={() =>
+                            params ? goto(route, params) : goto(route, undefined, true)
```

Comment:
self-nitpicking i know, but it could be done even better:

`goto(route, params, !params)` - preserve params only if `selectedAccount.params` are not set

please rename
`const { params } = useSelector(state => state.wallet.selectedAccount);`

remove line 66 

use `goto(route, params, !params)` in onClick block

Reply:
you can also revert `routeName` changes this way it will be 2 lines only fix :)